components/api/src/main/java/com/hotels/styx/api/Url.java

```diff
@@ -490,7 +490,7 @@ public static Builder url(String value) {
             return new Builder()
                     .scheme(uri.getScheme())
                     .authority(uri.getAuthority())
-                    .path(uri.getPath())
```

Comment:
This line is problematic. This decodes the path into a proper String, ie, `%2C -> ,`. But then, when you call toString on this URL, you get the decoded version instead of the raw one.

But, if this remains "raw", then the path remains already encoded. So when you call url.encodedUri()  on a path with encoded characters in it, you end up w/ encoded encoding characters.. ie, %D7 -> %25D7   This is because `%` encodes to `%25` 

Going to do what the JDK's URI does. Store both.